File: gui-client-web/src/toggle-collapse-action.ts
import { Action } from '@eclipse-glsp/client';

//Collapses or expands the subtree of a feature.
//If no elementId is given, the server uses the currently selected feature.

export interface ToggleCollapseAction extends Action {
    kind: typeof ToggleCollapseAction.KIND;
    elementId?: string;
}

export namespace ToggleCollapseAction {
    export const KIND = 'toggleCollapse';

    export function create(elementId?: string): ToggleCollapseAction {
        return { kind: KIND, elementId };
    }
}

File: gui-server/src/main/java/de/featjar/gui/action/ToggleCollapseAction.java
package de.featjar.gui.action;

import org.eclipse.glsp.server.actions.Action;

public class ToggleCollapseAction extends Action {

    public static final String KIND = "toggleCollapse";

    private String elementId;

    public ToggleCollapseAction() {
        super(KIND);
    }

    public String getElementId() {
        return elementId;
    }

    public void setElementId(final String elementId) {
        this.elementId = elementId;
    }
}
